extras/fuzz_crawl.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+import sys
+import atheris
+import scrapy
+from scrapy.crawler import CrawlerProcess
+from scrapy.exceptions import CloseSpider
+
+
+class test_spider(scrapy.Spider):
+    start_urls = ["http://google.com", "http://youtube.com/"]
+
+    def parse(self, response):
+        pass
+
+
+def TestOneInput(data):
+    fdp = atheris.FuzzedDataProvider(data)
+    test = fdp.ConsumeUnicodeNoSurrogates(fdp.ConsumeIntInRange(0, 4096))
+
+    try:
+        process = CrawlerProcess(settings={test})
+        process.crawl(test_spider)
+        process.start()
+    except CloseSpider:
+        pass
```

Comment:
I feel a bit out of my depth here. Not only I am not entirely sure how fuzzing works here, but I also don’t what will prevent those `start_urls` from actually being called, which is probably undesirable, or how passing the set `{test}` as the value of a parameter that expects a dict can work.

cc @wRAR

Reply:
Yes, I guess it should be a dict. Sorry about that.
I will close this PR and add better fuzz targets soon.